apps/dashboard/src/components/subscribers/locale-select.tsx

```diff
@@ -1,71 +1,129 @@
-      <SelectContent className="p-1">
-        {locales.map((item) => (
-          <SelectItem key={item.langIso} value={item.langIso} className="px-2">
-            <div className="flex w-full items-center gap-2">
-              <div>
-                <CountryFlags name={item.alpha2} className="size-4" />
-              </div>
-              <TruncatedText className="text-sm">
-                {item.langIso} - {item.langName}
-              </TruncatedText>
-            </div>
-          </SelectItem>
-        ))}
-      </SelectContent>
-    </Select>
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent className="w-[300px] rounded-lg border-t-0 p-0">
+        <Command>
+          <CommandInput
+            placeholder="Search locale..."
+            /**
+             * Scroll to top bug workaround: https://github.com/pacocoursey/cmdk/issues/233#issuecomment-2015998940
+             */
+            onValueChange={() => {
+              // clear pending scroll
+              clearTimeout(scrollId.current);
+
+              // the setTimeout is used to create a new task
+              // this is to make sure that we don't scroll until the user is done typing
+              // you can tweak the timeout duration ofc
+              scrollId.current = setTimeout(() => {
+                // inside your list select the first group and scroll to the top
+                const div = listRef.current;
+                div?.scrollTo({ top: 0, behavior: 'smooth' });
+              }, 0);
+            }}
+          />
+          <CommandList ref={listRef}>
+            <CommandEmpty>No locale found.</CommandEmpty>
+            <CommandGroup className="rounded-md py-2">
+              {locales.map((item) => (
+                <FlagItem
+                  countryCode={item.alpha2}
+                  languageName={item.langName}
+                  optionValue={item.langIso}
+                  key={item.langIso}
+                  onChange={(val) => {
+                    onChange(val);
+                    setOpen(false);
+                  }}
+                  currentValue={value}
+                />
+              ))}
+            </CommandGroup>
+          </CommandList>
+        </Command>
+      </PopoverContent>
+    </Popover>
   );
 }
+
+const FlagItem = ({
+  countryCode,
+  languageName,
+  optionValue,
+  onChange,
+  currentValue,
+}: {
+  countryCode: string;
+  languageName: string;
+  optionValue: string;
+  onChange: (val: string) => void;
+  currentValue?: string;
+}) => {
+  const CurrentFlag = countryCode ? flags[countryCode as Country] : RiEarthLine;
+  const isSelected = optionValue === currentValue;
+
+  return (
+    <CommandItem
+      className={cn('cursor-pointer gap-3', {
+        'bg-accent': isSelected,
+      })}
+      onSelect={() => onChange(optionValue)}
+    >
+      <div className="flex w-full items-center gap-2">
+        <div>{CurrentFlag && <CurrentFlag className="size-4" title={countryCode} />}</div>
```

Comment:
Consider avoid having an empty div.

```suggestion
        { CurrentFlag && <div><CurrentFlag className="size-4" title={countryCode} /></div>}
```